internal/controller/workspace/workspace.go

```diff
@@ -159,40 +159,50 @@ func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.E
 		return nil, errors.Wrap(err, errGetPC)
 	}
 
-	switch cr.Spec.ForProvider.Source {
-	case v1alpha1.ModuleSourceRemote:
-		// NOTE(ytsarev): Retrieve .git-credentials from Spec to /tmp outside of workspace directory
-		gitCredDir := filepath.Clean(filepath.Join("/tmp", dir))
-		if err := c.fs.MkdirAll(gitCredDir, 0700); err != nil {
-			return nil, errors.Wrap(err, errWriteGitCreds)
+	gitCredDir := filepath.Clean(filepath.Join("/tmp", dir))
+	if err := c.fs.MkdirAll(gitCredDir, 0700); err != nil {
+		return nil, errors.Wrap(err, errWriteCreds)
+	}
+
+	c.logger.Info("downloading credential files for module", "module", cr.Spec.ForProvider.Module)
+
+	for _, cd := range pc.Spec.Credentials {
+		data, err := resource.CommonCredentialExtractor(ctx, cd.Source, c.kube, cd.CommonCredentialSelectors)
+		if err != nil {
+			return nil, errors.Wrap(err, errGetCreds)
 		}
-		for _, cd := range pc.Spec.Credentials {
-			if cd.Filename != gitCredentialsFilename {
-				continue
-			}
-			data, err := resource.CommonCredentialExtractor(ctx, cd.Source, c.kube, cd.CommonCredentialSelectors)
-			if err != nil {
-				return nil, errors.Wrap(err, errGetCreds)
-			}
-			p := filepath.Clean(filepath.Join(gitCredDir, filepath.Base(cd.Filename)))
-			if err := c.fs.WriteFile(p, data, 0600); err != nil {
-				return nil, errors.Wrap(err, errWriteGitCreds)
-			}
-			// NOTE(ytsarev): Make go-getter pick up .git-credentials, see /.gitconfig in the container image
-			err = os.Setenv("GIT_CRED_DIR", gitCredDir)
-			if err != nil {
-				return nil, errors.Wrap(err, errRemoteModule)
+
+		credDir, credFile := filepath.Split(cd.Filename)
+		c.logger.Debug("processing credintial file", "file", credFile, "dir", credDir)
+
+		if credDir != "" {
+			if err = c.fs.MkdirAll(filepath.Clean(filepath.Join(gitCredDir, credDir)), 0700); err != nil {
+				return nil, errors.Wrap(err, errWriteCreds)
 			}
 		}
+		p := filepath.Clean(filepath.Join(gitCredDir, credDir, credFile))
+		if err := c.fs.WriteFile(p, data, 0600); err != nil {
+			return nil, errors.Wrap(err, errWriteCreds)
+		}
+	}
+
+	err := os.Setenv("GIT_CRED_DIR", gitCredDir)
+	if err != nil {
+		return nil, errors.Wrap(err, errRemoteModule)
+	}
+
+	c.logger.Info("Downloading terraform module", "module", cr.Spec.ForProvider.Module, "source", cr.Spec.ForProvider.Source)
 
+	switch cr.Spec.ForProvider.Source {
+	case v1alpha1.ModuleSourceRemote:
 		client := getter.Client{
 			Src: cr.Spec.ForProvider.Module,
 			Dst: dir,
 			Pwd: dir,
 
 			Mode: getter.ClientModeDir,
 		}
-		err := client.Get()
+		err = client.Get()
 		if err != nil {
```

Comment:
```suggestion
		if err := client.Get(); err != nil {
```

Can we more narrowly scope this error, as suggested?